### DISCUSSION TITLE
Mailer interface being enforced by merging the recipient and record

### DISCUSSION THREAD

**Discussion by kapso** [excid3/noticed#286] [Q&A]
I am curious why is the mailer interface being enforced by merging the recipient and record objects with the params. - https://github.com/excid3/noticed/blob/master/lib/noticed/delivery_methods/email.rb#L51

Why not let the mailer implementation decide what parameters it wants to receive?

**Comment by excid3** [comment]
The mailer always needs to know who the recipient is and what notification it is for, so we include it by default.

You can override and provide any parameters you want.

**Reply by kapso** [reply]
What if in my mailers, I am grabbing the recipient differently? And don't want it to be passed as a parameter.

**Reply by excid3** [reply]
Why use noticed then if you're going to ignore it's recipient? Just send the email without noticed.

**Reply by kapso** [reply]
Yea, that's a fair point, which then raises for us - why even use Noticed. 

The point is that the mailer framework is not part of Noticed, but the current Noticed design is forcing a specific interface on mailers, even after you override it with a `format` method. So essentially forcing to use mailers in a "Noticed" specific way.

**Reply by excid3** [reply]
Write your own delivery method if you don't want to use the built-in one. 👍
